### PR TITLE
Detect very long titles when no running title is specified

### PR DESCRIPTION
**Pull request by jwbos**

### CODE REVIEW

File: iacrcc/iacrcc.cls [hunk 1]
```diff
@@ -769,11 +769,15 @@
...
      \gdef\@plaintitle{\@IACR@title@plaintext}%
-     \expandafter\checkstring\expandafter{\@IACR@title@plaintext}{plaintext xxargument to title must be plain text}%
+     \expandafter\checkstring\expandafter{\@IACR@title@plaintext}{plaintext argument to title must be plain text}%
   \fi
   \gdef\@title{#2}%
   \if\relax\expandafter\detokenize\expandafter{\@IACR@title@running}\relax
     \gdef\@IACR@title@running{\@plaintitle}%
+    \StrLen{#2}[\IACR@tmp]%
+    \ifnumgreater{\IACR@tmp}{65}{%
+      \ClassError{iacrcc}{The title appears to be very long. Please define a running title by passing running=<runningtitle> to the \string\title\string macro}{}%
```
Comment (kmccurley) [inline iacrcc/iacrcc.cls:779]: I think this should be ClassWarning and not ClassError. We don't want to risk forcing people to shorten their title if they don't need to. It's possible that a 100 character title will still fit. The title in eprint 2023/1057 is 72 characters but this easily fits.
image
We should not stop compilation unless we are quite sure that there is something wrong.

**Review by kmccurley — COMMENTED** [review]
I really don't think we should use \ClassError, because we cannot be confident on how wide something will render.

I had another idea though that I think may be a better approach. We could use
```
\fancyhead[RE]{\makebox[.95\linewidth][l]{\@IACR@title@running}}
```
This forces the running head to be on a single line (in LR mode), and it constrains the width to 95% of the width of the line. If the running title is too long then it will hang out in the right margin and generate an overfull `\hbox` on every even page at the top. Many authors ignore overfull hboxes, but forcing the running title to be on a single line of maximal size should get the point across to authors.

**Comment by kmccurley** [comment]
After further investigation, the suggestion I used for using `\makebox` doesn't generate an hbox on every even page, but generates overfull vbox instead (and apparently just one overfull \hbox). If we use 
```
\fancyhead[RE]{\mbox{\@IACR@title@running}}
```
then it generates an overfull hbox and looks like:
image

This at least makes sure that it is written on a single line and makes it hang over both sides.

**Comment by jwbos** [comment]
> I really don't think we should use \ClassError, because we cannot be confident on how wide something will render.
> 
> I had another idea though that I think may be a better approach. We could use
> 
> ```
> \fancyhead[RE]{\makebox[.95\linewidth][l]{\@IACR@title@running}}
> ```
> 
> This forces the running head to be on a single line (in LR mode), and it constrains the width to 95% of the width of the line. If the running title is too long then it will hang out in the right margin and generate an overfull `\hbox` on every even page at the top. Many authors ignore overfull hboxes, but forcing the running title to be on a single line of maximal size should get the point across to authors.

One does not necessary exclude the other. This approach is in line with how we handle multiple authors. When more than five authors are used we throw an error and force the author to use the `\authorrunning` macro (even if the author names would not overflow). This is (imo) exactly the same: we are saying something is wrong but in case the title is above a certain threshold we ask the author to think and specify a running title to reduce the copy editing work.

File: iacrcc/iacrcc.cls [hunk 2]
```diff
@@ -769,11 +769,15 @@
...
   \else
      \gdef\@plaintitle{\@IACR@title@plaintext}%
-     \expandafter\checkstring\expandafter{\@IACR@title@plaintext}{plaintext xxargument to title must be plain text}%
+     \expandafter\checkstring\expandafter{\@IACR@title@plaintext}{plaintext argument to title must be plain text}%
   \fi
   \gdef\@title{#2}%
   \if\relax\expandafter\detokenize\expandafter{\@IACR@title@running}\relax
     \gdef\@IACR@title@running{\@plaintitle}%
+    \StrLen{#2}[\IACR@tmp]%
+    \ifnumgreater{\IACR@tmp}{65}{%
```
[the remark this answers is not in the document]
Comment (jwbos) [reply iacrcc/iacrcc.cls:778]: We could increase 65 to 70: fine by me :-)

File: iacrcc/iacrcc.cls (hunk 1 above)
Comment (jwbos) [reply iacrcc/iacrcc.cls:779]: I disagree. This is also how we handle `\authorrunning`.

File: iacrcc/iacrcc.cls (hunk 2 above)
[the remark this answers is not in the document]
Comment (jwbos) [reply iacrcc/iacrcc.cls]: This is changed to 70

File: iacrcc/iacrcc.cls (hunk 1 above)
Comment (jwbos) [reply iacrcc/iacrcc.cls]: Changed to Warning

**Review by kmccurley — APPROVED** [review]

**Pull request merged**